src/vs/workbench/api/browser/mainThreadLanguageFeatures.ts

```diff
@@ -302,6 +302,15 @@ export class MainThreadLanguageFeatures extends Disposable implements MainThread
 			}),
 			this._languageFeaturesService.multiDocumentHighlightProvider.register(selector, <languages.MultiDocumentHighlightProvider>{
 				provideMultiDocumentHighlights: (model: ITextModel, position: EditorPosition, otherModels: ITextModel[], token: CancellationToken): Promise<Map<URI, languages.DocumentHighlight[]>> => {
+					if (model.isDisposed()) {
```

Comment:
@Yoyokrazy I understand the check on line330 but this should never happen or did you really encounter that? It would be an issue with how we invoke the provider and the right fix wouldn't be here but in the invocation logic because nothing done here helps the monaco editor nor internal API users

Reply:
https://github.com/microsoft/vscode/pull/197583

Reply:
Looked this over, approved since it looks like Aaron's machine wasn't hitting the error anymore. Continuing to keep an eye on error telemetry to see if there are any other odd cases that arise.